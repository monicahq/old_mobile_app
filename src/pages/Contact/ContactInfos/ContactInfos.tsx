import React, {PureComponent} from 'react';
import SegmentedControlTab from 'react-native-segmented-control-tab';

import {I18n} from '@i18n';
import {IContact} from '@models';
import {primaryColor, segmentedControlGroupStyles} from '@theme';
import {Contact} from './Contact';
import {Family} from './Family';
import {Meet} from './Meet';
import {Work} from './Work';

const values = [
  I18n.t('contacts:family.family'),
  I18n.t('contacts:meet.how'),
  I18n.t('contacts:work.work'),
  I18n.t('contacts:contact.contact'),
];

interface IContactInfosProps {
  contact: IContact;
}
interface IContactInfosState {
  index: number;
}

export class ContactInfos extends PureComponent<
  IContactInfosProps,
  IContactInfosState
> {
  public state = {
    index: 2,
  };
  public setSegmentedControlState = index => this.setState({index});
  public render() {
    const {index} = this.state;
    const {contact} = this.props;

    const Elem =
      index === 0 ? Family : index === 1 ? Meet : index === 2 ? Work : Contact;

    return [
      <SegmentedControlTab
        key={0}
        tabStyle={segmentedControlGroupStyles.tabStyle}
        tabTextStyle={segmentedControlGroupStyles.tabTextStyle}
        activeTabTextStyle={segmentedControlGroupStyles.activeTabTextStyle}
        activeTabStyle={segmentedControlGroupStyles.activeTabStyle}
        values={values}
        selectedIndex={index}
        onTabPress={this.setSegmentedControlState}
      />,
      <Elem key={1} contact={contact} />,
    ];
  }
}
