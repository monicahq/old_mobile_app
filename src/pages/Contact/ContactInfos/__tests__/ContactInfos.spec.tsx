import {contact1} from '@src/__mock__/contact';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {ContactInfos} from '../ContactInfos';

describe('Pages', () => {
  describe('Contact / ContactInfos', () => {
    it('should renders correctly', () => {
      expect(
        toJson(shallow(<ContactInfos contact={contact1} />))
      ).toMatchSnapshot();
    });
  });
});
