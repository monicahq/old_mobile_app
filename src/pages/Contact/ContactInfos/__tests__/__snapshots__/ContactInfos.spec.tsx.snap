// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Contact / ContactInfos should renders correctly 1`] = `
Array [
  <SegmentedControlTab
    accessibilityLabels={Array []}
    accessible={true}
    activeTabBadgeContainerStyle={Object {}}
    activeTabBadgeStyle={Object {}}
    activeTabOpacity={1}
    activeTabStyle={
      Object {
        "backgroundColor": "#325776",
      }
    }
    activeTabTextStyle={
      Object {
        "color": "white",
      }
    }
    allowFontScaling={true}
    badges={
      Array [
        "",
        "",
        "",
      ]
    }
    borderRadius={5}
    enabled={true}
    key="0"
    multiple={false}
    onTabPress={[Function]}
    selectedIndex={2}
    selectedIndices={
      Array [
        0,
      ]
    }
    tabBadgeContainerStyle={Object {}}
    tabBadgeStyle={Object {}}
    tabStyle={
      Object {
        "borderColor": "#325776",
      }
    }
    tabTextStyle={
      Object {
        "color": "#325776",
      }
    }
    tabsContainerDisableStyle={
      Object {
        "opacity": 0.6,
      }
    }
    tabsContainerStyle={Object {}}
    textNumberOfLines={1}
    values={
      Array [
        "Relations",
        "How you met",
        "Work info",
        "Contact",
      ]
    }
  />,
  <Work
    contact={
      Object {
        "addresses": Array [
          Object {
            "city": "Montreal",
            "street": "5815 Park",
          },
        ],
        "contactFields": Array [
          Object {
            "contact_field_type": Object {
              "fontawesome_icon": "fa fa-facebook",
              "name": "name",
              "protocol": "http",
              "type": "email",
            },
            "content": "test@gmail.com",
          },
        ],
        "first_name": "Theo",
        "id": 2,
        "information": Object {
          "avatar": Object {
            "default_avatar_color": "blue",
            "source": "external",
            "url": "testurl",
          },
          "career": Object {
            "company": "Google",
            "job": "Developer",
            "linkedin_profile_url": "http://mylinkedin",
          },
          "dates": Object {
            "birthdate": Object {
              "date": null,
            },
            "deceased_date": Object {
              "date": null,
            },
          },
          "how_you_met": Object {
            "first_met_date": Object {
              "date": "1991-12-17",
            },
            "first_met_through_contact": null,
            "general_information": "test",
          },
          "relationships": Object {
            "family": Object {
              "contacts": Array [],
              "total": 0,
            },
            "friend": Object {
              "contacts": Array [],
              "total": 0,
            },
            "love": Object {
              "contacts": Array [],
              "total": 0,
            },
            "work": Object {
              "contacts": Array [],
              "total": 0,
            },
          },
        },
        "last_name": "Mathieu",
        "statistics": Object {
          "number_of_activities": 2,
          "number_of_calls": 3,
          "number_of_debts": 4,
          "number_of_gifts": 5,
          "number_of_notes": 6,
          "number_of_reminders": 9,
          "number_of_tasks": 1,
        },
        "updated_at": "2018-12-17",
      }
    }
    key="1"
  />,
]
`;
